### PR TITLE
FortiOS modules for 2.9 - 1

### PR DESCRIPTION
**Pull request by migumun**
##### SUMMARY
This PR contains an update to fortios modules that includes all of the changes requested for 2.9.
This is the first group of 20 modules.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

fortios_antivirus_heuristic
fortios_antivirus_profile
fortios_antivirus_quarantine
fortios_antivirus_settings
fortios_application_custom
fortios_application_group
fortios_application_list
fortios_application_name
fortios_application_rule_settings
fortios_authentication_rule
fortios_authentication_scheme
fortios_authentication_setting
fortios_dlp_filepattern
fortios_dlp_fp_doc_source
fortios_dlp_fp_sensitivity
fortios_dlp_sensor
fortios_dlp_settings
fortios_dnsfilter_domain_filter
fortios_dnsfilter_profile
fortios_endpoint_control_client

### CODE REVIEW

File: lib/ansible/modules/network/fortios/fortios_antivirus_heuristic.py [hunk 1]
```diff
@@ -44,37 +41,48 @@
     - fortiosapi>=0.9.8
 options:
     host:
-       description:
-            - FortiOS or FortiGate ip address.
-       required: true
+        description:
+            - FortiOS or FortiGate IP address.
+        type: str
+        required: true
```
Comment (NilashishC) [inline lib/ansible/modules/network/fortios/fortios_antivirus_heuristic.py]: Argspec says `required: False`.

File: lib/ansible/modules/network/fortios/fortios_antivirus_heuristic.py [hunk 2]
```diff
@@ -44,37 +41,48 @@
...
     username:
         description:
             - FortiOS or FortiGate username.
+        type: str
         required: true
```
Comment (NilashishC) [inline lib/ansible/modules/network/fortios/fortios_antivirus_heuristic.py]: Same as above.

**Comment by NilashishC** [comment]
Overall LGTM. However, before merging, please remove the `required: True` from docs wherever not required and check the unit tests where we're passing `state` key even though its not in the arspec.

File: lib/ansible/modules/network/fortios/fortios_antivirus_heuristic.py (hunk 1 above)
Comment (migumun) [reply lib/ansible/modules/network/fortios/fortios_antivirus_heuristic.py]: Good spot. I'll fix it and submit a new commit

File: lib/ansible/modules/network/fortios/fortios_antivirus_heuristic.py (hunk 2 above)
Comment (migumun) [reply lib/ansible/modules/network/fortios/fortios_antivirus_heuristic.py]: Good spot. I'll fix it and submit a new commit

**Comment by mamunozgonzalez** [comment]
> Overall LGTM. However, before merging, please remove the `required: True` from docs wherever not required and check the unit tests where we're passing `state` key even though its not in the arspec.

Thanks a lot. Regarding 'state' keyword in the Unit Tests, its presence is nice to test the capability of the module to filter data before it is sent to FortiGate REST API, even in the cases where the data is not really part of the argspec. 
Please note that Ansible argspec will prevent the module from being executed if there is a keyword that is not included in the argspec and this unit test is testing the capability of the internal method to filter data and remove attributes whose value is 'None'.

**Pull request merged**